xds/internal/balancer/lrs/balancer.go

```diff
@@ -212,11 +212,11 @@ func (w *xdsClientWrapper) update(newConfig *LBConfig) error {
 		w.loadWrapper.UpdateClusterAndService(w.clusterName, w.edsServiceName)
 	}
 
-	if w.lrsServerName != newConfig.LoadReportingServerName {
+	if w.lrsServerName == nil || *w.lrsServerName != newConfig.LoadReportingServerName {
 		// LoadReportingServerName is different, load should be report to a
 		// different server, restart.
 		restartLoadReport = true
-		w.lrsServerName = newConfig.LoadReportingServerName
+		w.lrsServerName = &newConfig.LoadReportingServerName
```

Comment:
Should we just create a new string pointer instead? Taking the address of `newConfig.LoadReportingServerName` would mean that the `newConfig` struct cannot be GC'ed. Don't think its a big issue, but something to consider.

Reply:
I thought this pointer only keeps the string around, not the full struct?

This is `& (newConfig.LoadReportingServerName)`
not `& (newConfig) .LoadReportingServerName`

I tried to run gctrace, but didn't know how to read the output.

Reply:
You might be right. Only the underlying array used to store the string bytes might be kept around.